### PR TITLE
Add temporal types, randomUUID, typeOf, and improve type handling

### PR DESCRIPTION
**Pull request by barakb**
## Summary by CodeRabbit

* **New Features**
  * Support for temporal types: Datetime, Date, Time, Duration
  * New helpers: randomUUID(), tostringOrNull(), and typeOf()

* **Improvements**
  * tostring() now formats temporal types and Point
  * isEmpty() accepts strings, maps, lists, and null
  * Temporal values stored/indexed as numeric fields when applicable
  * Improved concatenation, serialization, and temporal formatting

* **Updates**
  * Display name changed from "Relationship" to "Edge"

* **Bug Fixes**
  * Safer numeric parsing for exponents and clearer error when setting missing variables

* **Tests**
  * Re-enabled tests for UUIDs, division edge cases, type errors, and typeOf

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by barakb** [comment]
@coderabbitai full review

**Comment by barakb** [comment]
@coderabbitai full review

File: graph/src/runtime/value.rs [hunk 1]
```diff
@@ -590,10 +663,14 @@ impl Value {
             Self::List(_) => String::from("List"),
             Self::Map(_) => String::from("Map"),
             Self::Node(_) => String::from("Node"),
-            Self::Relationship(_) => String::from("Relationship"),
+            Self::Relationship(_) => String::from("Edge"),
```
[the remark this answers is not in the document]
Comment (barakb) [reply graph/src/runtime/value.rs]: this is fine

**Review by AviAvni — APPROVED** [review]
